### PR TITLE
Lazy loading of zarr timestamps

### PR DESCRIPTION
**Pull request by alejoe91**
This was previously discussed here: https://github.com/SpikeInterface/spikeinterface/pull/2828

To sum up, loading and decompressing zarr timestamps for very long recordings can be quite time consuming, so we want to avoid doing that at init. When fetching the timestamps though, if they ar not a numpy array they are cast and cached as np.arryas, to avoid re-reading and re-decompressing at every call

### CODE REVIEW

File: src/spikeinterface/core/baserecording.py [hunk 1]
```diff
@@ -811,10 +811,9 @@ def __init__(self, sampling_frequency=None, t_start=None, time_vector=None):
 
     def get_times(self):
         if self.time_vector is not None:
-            if isinstance(self.time_vector, np.ndarray):
-                return self.time_vector
-            else:
-                return np.array(self.time_vector)
+            if not isinstance(self.time_vector, np.ndarray):
```
Comment (h-mayorquin) [inline src/spikeinterface/core/baserecording.py:814]: I think you can just always call np.asaray() which by default will just pass the data along if it is already and np.ndarray but will create a copy if it is hdf5, zarr or a memmap.

**Review by h-mayorquin — COMMENTED** [review]
Yes, I think it is better. Zarr is not the format were we want the data to be read when called. 

Can you add the typing to `get_times()` -> np.ndaarray, this will make it clear that is returning and in memory object and also we should add a docstring describing this behavior.

File: src/spikeinterface/core/baserecording.py (hunk 1 above)
Comment (h-mayorquin) [reply src/spikeinterface/core/baserecording.py:814]: Yes!

Comment (alejoe91) [reply src/spikeinterface/core/baserecording.py]: Then done in last commit :)

**Review by h-mayorquin — APPROVED** [review]

**Pull request merged**